python/ray/dashboard/modules/reporter/gpu_providers.py

```diff
@@ -149,6 +154,110 @@ def _shutdown(self):
 
     def get_gpu_utilization(self) -> List[GpuUtilizationInfo]:
         """Get GPU utilization information for all NVIDIA GPUs and MIG devices."""
+
+        return (
+            self._get_nvsmi_gpu_usage()
+            if self._using_nvidia_smi
+            else self._get_pynvml_gpu_usage()
+        )
+
+    def _get_nvsmi_gpu_usage(self) -> List[GpuUtilizationInfo]:
+        try:
+            gpu_info = subprocess.run(
+                [
+                    "nvidia-smi",
+                    "--query-gpu=index,name,uuid,utilization.gpu,memory.used,memory.total",
+                    "--format=csv,noheader,nounits",
+                ],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+            gpus = []
+            for line in gpu_info.stdout.strip().split("\n"):
+                index, name, uuid, util, mem_used, mem_total = line.split(", ")
+                gpus.append(
+                    GpuUtilizationInfo(
+                        index=int(index),
+                        name=name,
+                        uuid=uuid,
+                        utilization_gpu=int(util),
+                        memory_used=int(mem_used),
+                        memory_total=int(mem_total),
+                        processes_pids=[],
+                    )
+                )
+
+            processes_info = subprocess.run(
+                ["nvidia-smi", "pmon", "-c", "1"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                check=True,
+                text=True,
+            )
+            processes_info = self._parse_nvsmi_output(processes_info.stdout)
+            for gpu in gpus:
+                gpu_id = gpu["index"]
+                if gpu_id in processes_info:
+                    for pid, mem, sm in processes_info[gpu_id]:
+                        gpu["processes_pids"].append(
+                            ProcessGPUInfo(
+                                pid=int(pid),
+                                gpu_memory_usage=int(
+                                    mem * gpu["memory_total"] / 100
+                                ),  # MB
+                                gpu_utilization=int(sm),
+                            )
+                        )
+            return gpus
+        except subprocess.CalledProcessError as e:
+            logger.debug(
+                f"nvidia-smi failed to call: {e}. Is nvidia-smi installed? Falling back to pynvml."
+            )
+            self._using_nvidia_smi = False
+            return self._get_pynvml_gpu_usage()
+
+    @staticmethod
+    def _parse_nvsmi_output(nvsmi_stdout: str) -> Dict[int, List[Tuple[int, int, int]]]:
+        """Parse the output of nvidia-smi pmon -c 1."""
```

Comment:
Could you document the return type? What each field means. You may want to make it a struct or named tuple.